test/async-hooks/hook-checks.mjs

```diff
@@ -0,0 +1,53 @@
+import '../common/index.mjs';
+import assert, { ok, strictEqual } from 'assert';
```

Comment:
nit: since we are rewriting the test, should we use `node:` prefix for node modules? Go with whichever form you prefer.
```suggestion
import assert, { ok, strictEqual } from 'node:assert';
```